src/main/java/me/xmrvizzy/skyblocker/skyblock/dungeon/DungeonBlaze.java

```diff
@@ -20,92 +22,134 @@
-                        RenderUtils.drawBoxOutline(nextBlaze, outlineColorWhite, 5f);
-                        RenderHelper.renderLinesFromPoints(wrc, new Vec3d[] { blaze.getCenter(), nextBlaze.getCenter() }, WHITE_COLOR_COMPONENTS, 1f, 5f);
-                    }
+                    renderBlazeOutline(highestBlaze, nextHighestBlaze, wrc);
                 }
-
-                /* Outline */
                 if (lowestBlaze.getY() > 69) {
-                    Box blaze = lowestBlaze.getBoundingBox().expand(0.3, 0.9, 0.3).offset(0, -1.1, 0);
-                    RenderUtils.drawBoxOutline(blaze, outlineColorRed, 5f);
-                    
-                    if (nextLowestBlaze != null && nextLowestBlaze.isAlive() && nextLowestBlaze != lowestBlaze) {
-                        Box nextBlaze = nextLowestBlaze.getBoundingBox().expand(0.3, 0.9, 0.3).offset(0, -1.1, 0);
-                        RenderUtils.drawBoxOutline(nextBlaze, outlineColorWhite, 5f);
-                        RenderHelper.renderLinesFromPoints(wrc, new Vec3d[] { blaze.getCenter(), nextBlaze.getCenter() }, WHITE_COLOR_COMPONENTS, 1f, 5f);
-                    }
+                    renderBlazeOutline(lowestBlaze, nextLowestBlaze, wrc);
                 }
             }
         } catch (Exception e) {
-            LOGGER.warn("[Skyblocker BlazeRenderer] " + e);
-            e.printStackTrace();
+            handleException(e);
         }
     }
-}
\ No newline at end of file
+
+    /**
+     * Renders outlines for Blaze entities and connections between them.
+     *
+     * @param blaze     The Blaze entity for which to render an outline.
+     * @param nextBlaze The next Blaze entity for connection rendering.
+     * @param wrc       The WorldRenderContext used for rendering.
+     */
+    private static void renderBlazeOutline(ArmorStandEntity blaze, ArmorStandEntity nextBlaze, WorldRenderContext wrc) {
+        Box blazeBox = blaze.getBoundingBox().expand(0.3, 0.9, 0.3).offset(0, -1.1, 0);
+        RenderUtils.drawBoxOutline(blazeBox, DungeonBlaze.outlineColorGreen, 5f);
+
+        if (nextBlaze != null && nextBlaze.isAlive() && nextBlaze != blaze) {
+            Box nextBlazeBox = nextBlaze.getBoundingBox().expand(0.3, 0.9, 0.3).offset(0, -1.1, 0);
+            RenderUtils.drawBoxOutline(nextBlazeBox, DungeonBlaze.outlineColorWhite, 5f);
+
+            Vec3d blazeCenter = blazeBox.getCenter();
+            Vec3d nextBlazeCenter = nextBlazeBox.getCenter();
+
+            RenderHelper.renderLinesFromPoints(wrc, new Vec3d[]{blazeCenter, nextBlazeCenter}, WHITE_COLOR_COMPONENTS, 1f, 5f);
+        }
+    }
+
+    /**
+     * Handles exceptions by logging and printing stack traces.
+     *
+     * @param e The exception to handle.
+     */
+    private static void handleException(Exception e) {
+        LOGGER.warn("[Skyblocker BlazeRenderer] " + e);
+        e.printStackTrace();
```

Comment:
```suggestion
        LOGGER.warn("[Skyblocker Dungeon Blaze] Encountered an unknown exception", e);
```
Something like this would be better than `printStackTrace`.